lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3525,4 +3525,17 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Determines if a string literal is valid as an identifier in this
+     * platform.
+     *
+     * @param string $name The string literal to test
+     *
+     * @return bool
+     */
+    public function isValidIdentifier($name)
```

Comment:
I remember that I was trying to implement something like this some time ago and we had a long discussion about that internally. The problem here are two things:

1. AFAIK know there is no secure way in PHP to be 100% sure that a given value is a valid identifier with regards to the underyling platform. A complex regex might catch 99% but I doubt it can provide 100% safety. @beberlei was asking PHP internals about that topic as far as I remember and they gave a similar answer.

2. If we cannot be 100% sure that a given identifier is valid, such a method will be misleading to the consumers as they will rely on that and in a worst case scenario use it with unfiltered user input which opens up for SQL injection.

The only option I see to put such a method in here is to document security issue well (in the code and also in the documentation).

@beberlei @Ocramius thoughts?